crates/red_knot_python_semantic/resources/mdtest/attributes.md

```diff
@@ -440,12 +481,14 @@ reveal_type(C.pure_class_variable)  # revealed: Unknown
 
 C.pure_class_variable = "overwritten on class"
 
-# TODO: should be `Literal["overwritten on class"]`
+# TODO: should be  `Unknown | Literal["value set in class method"]` or
+# Literal["overwritten on class"]`, once/if we support local narrowing.
 # error: [unresolved-attribute]
 reveal_type(C.pure_class_variable)  # revealed: Unknown
 
 c_instance = C()
-# TODO: should be `Literal["overwritten on class"]`
+# TODO: should be `Literal["overwritten on class"]` once/if we support
+# local narrowing.
```

Comment:
interesting... do mypy/pyright support narrowing like this, where you narrow it on the class but then read it from an instance?

Reply:
Mypy doesn't seem to support this detection of implicit `pure_class_variable`s at all. Pyright supports narrowing in the case above (`C.pure_class_variable`), but not here.

Maybe I should remove the TODO here?

Reply:
Yeah, I don't think this is something we "have" to support -- I don't think it's something users will expect of us. (I think they _will_ expect attribute narrowing when the attribute is accessed on the same variable, such as `C.pure_class_variable` above.) Obviously if it's easy and naturally falls out of the implementation, then that's fantastic -- I think it isn't any more unsound than attribute narrowing in general. But I don't think it's something we need a TODO for.